### DISCUSSION TITLE
Semantic HTML-based charts and data visualizations

### DISCUSSION THREAD

**Discussion by sknep** [uswds/uswds#5381] [Ideas]
@iamjolly and I are interested in building some semantically-correct HTML-based chart components for the USWDS, similar to how ChartsCSS has demonstrated: https://chartscss.org/charts/bar/ but with USWDS design tokens and class naming conventions. @mgifford has started a related issue here: https://github.com/uswds/uswds/issues/5281 about proper attributes for SVG-based data visualizations, which is appropriate for complex graphs. 

At the same time, we believe some graphs can be accommodated via HTML and CSS, which offers a direct and accessible experience for those using various assistive technologies, with the added bonus of not needing to 'draw' the chart in a vector program. The [2021 10x Impact Report](https://10x.gsa.gov/reports/fy21-impact-report/#chartTitle) is an example of how we've rebuilt an SVG chart using HTML and data from a simple list (array), displayed as a bar chart using CSS. We'd like to propose a series of simple charts in this fashion as component variants to the USWDS. Robert and I are starting to work on this but would welcome other input and feedback!

**Comment by mgifford** [comment]
Very cool @sknep & @iamjolly. Worth looking at https://css-tricks.com/making-charts-with-css/ too

I love the example you provided too. Really, no need to overcomplicate it with a SVG if you can use some other structured content which is then marked up to visually provide a compelling image. 

Thanks for working on this. CSS is just so powerful. 

It would be so nice to have a set of CSS styles that could be used to help visualize data in a consistent, mobile-friendly, performant and accessible way.

**Reply by mejiaj** [reply]
Semantically-correct HTML-based charts make sense to me. What types of styles would be most useful? What challenges are you seeing with the existing styles i.e. _where are the gaps_ ?

**Reply by mgifford** [reply]
It would be really useful to actually have some sort of survey about the types of graphs/charts that appear on .gov sites right now. That baseline would be quite useful. I'm quite sure most of them wouldn't actually meet Section 508. But it would be useful to know what information they are trying to convey.
